server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1738,6 +1741,63 @@ public static int calculateNumRoutingShards(int numShards, Version indexVersionC
         return numShards * 1 << numSplits;
     }
 
+    /**
+     * Validates that the settings do not conflict with each other.
```

Comment:
nitpick here and in the other method, define which settings, "Validates that auto expand and search replicas are mutually exclusive."